src/App.js

```diff
@@ -23,49 +24,101 @@ const generateSquares = () => {
   }
 
   return squares;
-}
+};
 
 const App = () => {
 
   // This starts state off as a 2D array of JS objects with
   // empty value and unique ids.
   const [squares, setSquares] = useState(generateSquares());
+  const [isXTurn, setisXTurn] = useState(true);
+  const [isWinner, setIsWinner] = useState('')
+  // const [resetSquares, setResestSquares] = useState(squares)
 
   // Wave 2
   // You will need to create a method to change the square 
   //   When it is clicked on.
   //   Then pass it into the squares as a callback
+  const onClickCallback = (id) => {
+    const updatedSquares = [...squares];
+    for (let i=0; i<updatedSquares.length; i++) {
+      for (let j=0; j<updatedSquares[i].length; j++) {
+        if (updatedSquares[i][j].value && updatedSquares[i][j].id===id) {
+            return;
+        
+      }else if (isXTurn && updatedSquares[i][j].id === id) {
+          updatedSquares[i][j].value = PLAYER_1
+          setisXTurn(false)
+        } else if (updatedSquares[i][j].id === id) {
+          updatedSquares[i][j].value = PLAYER_2
+          setisXTurn(true)
+        }
+
+      }
+    }
+      setSquares(updatedSquares);
+      let winner = checkForWinner(updatedSquares);
+      setIsWinner(winner)
+  };
+    
+
+ 
 
 
-  const checkForWinner = () => {
+  const checkForWinner = (squares) => {
```

Comment:
`squares` is already in scope (this is a function within `App`) so you don't need to pass it in:

```suggestion
  const checkForWinner = () => {
```